stor/tests/test_dx.py

```diff
@@ -705,52 +828,898 @@ class TestTempUrl(DXTestCase):
+        self.setup_temporary_project()
+        dx_p = DXPath('dx://RandomProject:/')
+        with pytest.raises(exceptions.NotFoundError, match='Found no projects'):
+            dx_p.rmtree()
+
+
+class TestMakedirsP(DXTestCase):
+    def test_makedirs_p_project(self):
+        with pytest.raises(ValueError, match='Cannot create a project'):
+            DXPath('dx://project:').makedirs_p()
+
+    def test_makedirs_p_project_exists(self):
+        self.setup_temporary_project()
+        dx_p = DXPath('dx://' + self.project)
+        dx_p.makedirs_p()
+        self.assertTrue(dx_p.exists())
+
+    def test_makedirs_p_folder(self):
+        self.setup_temporary_project()
+        dx_p = DXPath('dx://' + self.project + ':/temp_folder')
+        dx_p.makedirs_p()
+        proj_path = DXPath('dx://' + self.project)
+        result = proj_path.listdir()
+        self.assert_dx_lists_equal(result, [dx_p])
+
+    def test_makedirs_p_nested_folder(self):
+        self.setup_temporary_project()
+        dx_p = DXPath('dx://' + self.project + ':/temp_folder/nested_folder')
+        dx_p.makedirs_p()
+        folder_path = DXPath('dx://' + self.project + ':/temp_folder')
+        proj_path = DXPath('dx://' + self.project)
+        result = proj_path.listdir()
+        self.assert_dx_lists_equal(result, [folder_path])
+        result = folder_path.listdir()
+        self.assert_dx_lists_equal(result, [dx_p])
+
+    def test_makedirs_p_folder_exists(self):
+        self.setup_temporary_project()
+        dx_p = DXPath('dx://' + self.project + ':/temp_folder')
+        dx_p.makedirs_p()
+        proj_path = DXPath('dx://' + self.project)
+        result = proj_path.listdir()
+        self.assert_dx_lists_equal(result, [dx_p])
+        dx_p = DXPath('dx://' + self.project + ':/temp_folder/nested_folder')
+        dx_p.makedirs_p()
+        folder_path = DXPath('dx://' + self.project + ':/temp_folder')
+        result = folder_path.listdir()
+        self.assert_dx_lists_equal(result, [dx_p])
+
+
+class TestDownloadObjects(DXTestCase):
+    def test_local_paths(self):
+        self.setup_temporary_project()
+        self.setup_file('/temp_folder/file1.txt')
+        self.setup_file('/temp_folder/file2.txt')
+        self.setup_file('/temp_folder/folder/file3.txt')
+        dx_p = DXPath('dx://' + self.project + ':/temp_folder')
+        posix_folder_p = Path('./{test_folder}'.format(
+            test_folder=self.project))
+        self.addCleanup(posix_folder_p.rmtree)
+        r = dx_p.download_objects(self.project,
+                                  ['file1.txt',
+                                   'file2.txt',
+                                   'folder/file3.txt'])
+        self.assertEquals(r, {
+            'file1.txt': self.project + '/file1.txt',
+            'file2.txt': self.project + '/file2.txt',
+            'folder/file3.txt': self.project + '/folder/file3.txt'
+        })
+
+    def test_absolute_paths(self):
```

Comment:
can this be combined with swift test case? (actually for 30s check here)

Reply:
No, the swift test uses mocks as well as sets download parameters by mocking settings. Plus, Idk how you would combine the two (maybe into shared_obs.py but it is way overkill). 

Also, what do you mean by 30s check here? Since the fixtures are included, this is instantaneous.

Reply:
cool sounds good